src/test/java/io/github/oliviercailloux/y2018/apartments/valuefunction/ApartmentValueFunctionTest.java

```diff
@@ -46,25 +49,113 @@ void apartmentValueFunctionTest() throws NumberFormatException {
 		BooleanValueFunction wifiV = new BooleanValueFunction(true);
 		valueFunction.setWifiValueFunction(wifiV);
 
-		LinearValueFunction nbBathroomsV = new LinearValueFunction(2, 3);
+		LinearValueFunction nbBathroomsV = new LinearValueFunction(2d, 3d);
 		valueFunction.setNbBathroomsValueFunction(nbBathroomsV);
 
-		LinearValueFunction floorAreaTerraceV = new LinearValueFunction(30, 50);
+		LinearValueFunction floorAreaTerraceV = new LinearValueFunction(30d, 50d);
 		valueFunction.setFloorAreaTerraceValueFunction(floorAreaTerraceV);
+	}
 
-		assertEquals(0.5, valueFunction.getSubjectiveValue(a), 0.0001);
+	/**
+	 * Function to test the some setters
+	 */
+	@Test
+	void checkValue() {
 
-		valueFunction.setTeleSubjectiveValueWeight(10);
+		LinearValueFunction lvf = (LinearValueFunction) valueFunction.getNbSleepingValueFunction();
+		assertEquals(5d, lvf.getInterval().upperEndpoint());
+		lvf = (LinearValueFunction) valueFunction.getNbBedroomsValueFunction();
+		assertEquals(4d, lvf.getInterval().upperEndpoint());
+		lvf = (LinearValueFunction) valueFunction.getNbBathroomsValueFunction();
+		assertEquals(3d, lvf.getInterval().upperEndpoint());
 
+	}
+
+	/**
+	 * Function to test the computing of the subjective value of an apartment
+	 */
+	@Test
+	void apartmentValueFunctionTest() {
+
+		assertEquals(0.5, valueFunction.getSubjectiveValue(a), 0.0001);
+
+		valueFunction.setTeleSubjectiveValueWeight(10d);
+		assertEquals(10d, valueFunction.getSubjectiveValueWeight(Criterion.TELE));
 		assertEquals(0.04587, valueFunction.getSubjectiveValue(a), 0.00001);
```

Comment:
Il serait sympathique d’indiquer en commentaire comment on peut vérifier à la main les valeurs attendues, et de vous assurer au passage que la AVF et les PVF sont suffisamment simples pour qu’on puisse faire le calcul de tête, autant que possible (ce qui, vu le résultat actuel de 0.04587, ne me parait pas être assurément le cas actuellement). En outre, ce serait un excellent exercice pour vérifier votre compréhension des VF, et une doc utile pour les équipes qui vous succéderont. FUTURE

Reply:
Opening an issue concerning this comment : [https://github.com/av1m/Apartments/issues/23](url)
@GabG02 I can fix this issue on iteration 2 if you want